src/consensus/block_proposer.hpp

```diff
@@ -96,7 +96,7 @@ class BlockProposer : public std::enable_shared_from_this<BlockProposer> {
   void start();
   void stop();
   std::shared_ptr<BlockProposer> getShared();
-  void setNetwork(std::weak_ptr<Network> network) { network_ = move(network); }
+  void setNetwork(std::shared_ptr<Network> network) { network_ = move(network); }
```

Comment:
are you sure about this one ? We have some weak_ptr's because of shared_ptr's circular dependency issue.

Reply:
I think in some commit after I returned it back, but forgot to include it during the splitting. Will check that

Reply:
are you sure then about the splitting then ?

Reply:
I think it is better in separate PR's but dont want to merge fixes with bugs...

Reply:
It wont have conflicts because it is just have the commit from other one PR and it will be removed after it is merge. This splitting is done to made a review easier and PRs are merging to a feature, not a develop. So after all  that merges I will just compare all the changes to original one I made